### PR TITLE
Added endpoints for registration and logging in

### PR DESCRIPTION
**Pull request by Croitoru**
+ token creation

### CODE REVIEW

File: ProjectR/TruEM/TruEM/Controllers/AuthController.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                }
+            }
+
+            var loggedInUser = new LoggedInUser
+            {
+                Username = user.Username,
+                Token = CreateToken(user)
+            };
+
+            return loggedInUser;
```
Comment (IRailean) [inline ProjectR/TruEM/TruEM/Controllers/AuthController.cs:82]: Here you can write directly:
return new LoggedInUser
            {
                Username = user.Username,
                Token = CreateToken(user)
            };

File: ProjectR/TruEM/TruEM/Controllers/AuthController.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly DataContext _context;
+        private readonly SymmetricSecurityKey _key;
+
+        public AuthController(DataContext context)
+        {
+            _key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes("MySuperSecretKey123"));
```
Comment (IRailean) [inline ProjectR/TruEM/TruEM/Controllers/AuthController.cs:24]: Suggestion:
Secret keys must be stored in a secure location. 
You can put it in appsettings.Development.json and read it from there.
A better approach is to store this key in the environment variable on your machine.

Such a key must not be exposed in a GitHub repo

File: ProjectR/TruEM/TruEM/Controllers/AuthController.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            };
+
+            var credentials = new SigningCredentials(_key, SecurityAlgorithms.HmacSha512Signature);
+
+            var tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(claims),
+                Expires = DateTime.Now.AddDays(1),
+                SigningCredentials = credentials,
+                Issuer = "https://localhost:5001"
```
Comment (IRailean) [inline ProjectR/TruEM/TruEM/Controllers/AuthController.cs:99]: Issuer IP should be moved to a config file.

File: ProjectR/TruEM/TruEM/Entity/AppUser.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace TruEM.Entity
+{
+    [Table(nameof(AppUser))]
+    public class AppUser
+    {
+        [Key]
+        public int Id { get; set; }
+        public string Username { get; set; }
+        public byte[] Password { get; set; }
```
Comment (IRailean) [inline ProjectR/TruEM/TruEM/Entity/AppUser.cs:16]: It is PasswordHash, not Password. Naming is important here.

File: ProjectR/TruEM/TruEM/Controllers/AuthController.cs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+            using var hmac = new HMACSHA512();
+            var appUser = new AppUser
+            {
+                Username = registerUserRequest.Username.ToLower(),
+                PasswordHash = hmac.ComputeHash(Encoding.UTF8.GetBytes(registerUserRequest.Password)),
+                PasswordSalt = hmac.Key
+            };
+
+            _context.AppUsers.Add(appUser);
```
Comment (IRailean) [inline ProjectR/TruEM/TruEM/Controllers/AuthController.cs:44]: AddAsync

File: ProjectR/TruEM/TruEM/Controllers/AuthController.cs (hunk 1 above)
Comment (Croitoru) [reply ProjectR/TruEM/TruEM/Controllers/AuthController.cs]: Implemented

File: ProjectR/TruEM/TruEM/Controllers/AuthController.cs (hunk 2 above)
Comment (Croitoru) [reply ProjectR/TruEM/TruEM/Controllers/AuthController.cs]: Done

File: ProjectR/TruEM/TruEM/Controllers/AuthController.cs (hunk 3 above)
Comment (Croitoru) [reply ProjectR/TruEM/TruEM/Controllers/AuthController.cs]: Done

**Pull request merged**